windows/winrm-gen.ps1

```diff
@@ -0,0 +1,83 @@
+param (
+    [string]$ComputerName = "servername.domain.com"
+)
+
+$FirewallParam = @{ 
+    DisplayName = 'Windows Remote Management (HTTPS-In)'
+    Direction = 'Inbound' 
+    LocalPort = 5986 
+    Protocol = 'TCP' 
+    Action = 'Allow' 
+    Program = 'System' 
+} 
+
+$ErrorActionPreference = "Stop"
+
+#
+# WinRM prechecks
+#
+try {
+    Write-Output "Checking if WinRM is enabled"
+    Test-WSMan > $null
+    Write-Output "Checking if WinRM configured"
+    winrm get winrm/config
+    if ( $LASTEXITCODE ) {
+        Write-Output "WinRM not configured, configuring with the default settings"
+        winrm quickconfig -transport:https
+    }
+}
+catch [System.Management.Automation.CommandNotFoundException] {
+    "winrm not found"
+    exit
+}
+catch {
+    "An error occurred that could not be resolved."
+    exit
```

Comment:
If you had to catch the error and then continue, these catches would make perfect sense, but you just exit. Therefore, please remove the try-catch block, as it may just hide the actual error message we might need for troubleshooting in case something goes wrong in that try block.